pkg/controllers/nodeclaim/lifecycle/launch.go

```diff
@@ -61,7 +62,22 @@ func (l *Launch) Reconcile(ctx context.Context, nodeClaim *v1beta1.NodeClaim) (r
 		created, err = l.linkNodeClaim(ctx, nodeClaim)
 	} else {
 		created, err = l.launchNodeClaim(ctx, nodeClaim)
+		npKey := nodepoolutil.Key{Name: nodeClaim.Labels[v1beta1.NodePoolLabelKey], IsProvisioner: nodeClaim.IsMachine}
```

Comment:
We shouldn't add the NodePool status condition update here since this controller is intended for the NodeClaim reconciliation. We basically need a NodePool controller that reconciles and checks the status condition of the `nodeClaimRef` and use that reconcile result to update the `NodeClassReady` condition which should further update the top-level `Ready` condition